src/cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -744,6 +748,37 @@ GpiIterator::Status VpiSingleIterator::next_handle(std::string &name,
         return GpiIterator::NOT_NATIVE;
 }
 
+GpiIterator::Status VpiPackageIterator::next_handle(std::string &,
+                                                    GpiObjHdl **hdl, void **) {
+    GpiObjHdl *new_obj;
+    vpiHandle obj;
+
+    if (NULL == m_iterator) return GpiIterator::END;
+
+    while (true) {
+        obj = vpi_scan(m_iterator);
+        if (NULL == obj) return GpiIterator::END;
+
+        PLI_INT32 type = vpi_get(vpiType, obj);
+        if (type == vpiPackage) break;
```

Comment:
If the type isn't `vpiPackage` when we've asked it to iterate over package, that's an error condition. Log an error message.